src/main/java/net/minecraftforge/resource/ResourceCacheManager.java

```diff
@@ -413,6 +446,178 @@ public boolean cacheLoaded()
+        /**
+         * The path to the boot configuration file.
+         */
+        private static final Path CONFIG_PATH = FMLPaths.CONFIGDIR.get().resolve("forge-resource-caching.toml");
+        /**
+         * The current configuration in the file.
+         * Each entry in the list represents a single line, might be a comment or a configuration entry.
+         * This is an atomic reference since the file watchdog runs on a different thread and the list might be updated while we are reading it.
+         */
+        private static final AtomicReference<List<String>> CURRENT_CONFIG_CONTENTS = new AtomicReference<>(readConfigFileContent());
+
+        /**
+         * Creates a new instance of the handler.
+         * Registers the watchdog thread.
+         */
+        private ResourceManagerBootCacheConfigurationHandler()
+        {
+            //Create and register the watchdog thread.
+            final Thread watchDog = new Thread(ResourceManagerBootCacheConfigurationHandler::monitorBootConfiguration);
+            watchDog.setDaemon(true); //Background thread.
+            watchDog.setName("Forge Resource Cache Configuration Watchdog");
+            watchDog.start();
+        }
+
+        /**
+         * Sets up, and runs a monitor on the boot configuration file, on the current thread.
+         * Make sure to invoke this from a background thread.
+         */
+        private static void monitorBootConfiguration()
+        {
+            //Get the configuration directory.
+            Path path = CONFIG_PATH.getParent();
+            try (final WatchService watchService = FileSystems.getDefault().newWatchService())
+            {
+                //Register a watch service on the configuration directory for any changes.
+                path.register(watchService, ENTRY_CREATE, ENTRY_MODIFY, ENTRY_DELETE);
+                while (true)
+                {
+                    //Grab the next key
+                    WatchKey key = watchService.take();
+                    //For each event in that key we check if it is our config file that is changed.
+                    for (WatchEvent<?> event : key.pollEvents())
+                    {
+                        if (event.context().equals(path.relativize(CONFIG_PATH)))
+                        {
+                            //And then we atomically update the config contents.
+                            CURRENT_CONFIG_CONTENTS.set(readConfigFileContent());
+                        }
+                    }
+                    //Reset the key, and continue.
+                    key.reset();
+                }
+            }
+            catch (IOException | InterruptedException e)
+            {
+                //Some generic failure occurred, that is not good. Let's kill it!
+                throw new RuntimeException("Failed to read boot configuration contents on update.", e);
+            }
+        }
```

Comment:
This can be changed to use the nightconfig-provided `FileWatcher`, reinventing the wheel is not needed.